### PR TITLE
Add a new `update available` widget to statusBar

### PR DESCRIPTION
**Pull request by knolleary**
This is a continuation of #4935 that replaces that PR.

It refactors the palette manager code to allow us to reload the catalogues *without* generating the whole palette manager UI. This makes it more efficient to do background checks of updates so we can keep the status widget up to date.

### CODE REVIEW

**Comment by knolleary** [comment]
Sigh - pushed the brnach before I had commited all my work to it.... update incoming....

**Review by GogoVega — CHANGES_REQUESTED** [review]
Overall it seems good to me 🙂, there is just a problem with `catalogueLoadStart`.

File: packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js [hunk 1]
```diff
@@ -444,37 +471,18 @@ RED.palette.editor = (function() {
             packageList.editableList('empty');
 
             $(".red-ui-palette-module-shade-status").text(RED._('palette.editor.loading'));
```
[the remark this answers is not in the document]
Comment (knolleary) [reply packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js:454]: Fixed

File: packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js [hunk 2]
```diff
@@ -421,9 +451,6 @@ RED.palette.editor = (function() {
         } else {
             catalogueLoadErrors = true;
         }
-        if (catalogueCount > 1) {
-            $(".red-ui-palette-module-shade-status").html(RED._('palette.editor.loading')+"<br>"+catalogueLoadStatus.length+"/"+catalogueCount);
-        }
         if (catalogueLoadStatus.length === catalogueCount) {
             if (catalogueLoadErrors) {
                 RED.notify(RED._('palette.editor.errors.catalogLoadFailed',{url: url}),"error",false,8000);
```
[the remark this answers is not in the document]
Comment (knolleary) [reply packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js]: I've simplified the error reporting. Showing a notification at all isn't that helpful as the user can't do anything about it. We already log the error details to the browser console. I think that is good enough.

File: packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js [hunk 3]
```diff
@@ -385,54 +385,62 @@ RED.palette.editor = (function() {
...
+        loadedIndex = {}
+        loadedCatalogs.length = 0
+        let handled = 0
+        for (let index = 0; index < catalogues.length; index++) {
+            const url = catalogues[index];
+            $.getJSON(url, {_: new Date().getTime()},function(v) {
+                loadedCatalogs.push({ index: index, url: url, name: v.name, updated_at: v.updated_at, modules_count: (v.modules || []).length })
+                handleCatalogResponse({ url: url, name: v.name},index,v);
+            }).fail(function(jqxhr, textStatus, error) {
+                console.warn("Error loading catalog",url,":",error);
```
Comment (GogoVega) [inline packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js:404]:
```suggestion
                $(".red-ui-palette-module-shade-status").text(RED._('palette.editor.errors.catalogLoadFailed'));
                console.warn("Error loading catalog",url,":",error);
```

Comment (GogoVega) [reply packages/node_modules/@node-red/editor-client/src/js/ui/palette-editor.js:404]: Not useful. I don't know if this case happens often.
We'll leave it like that for now.

**Pull request merged**